client/X11/xf_window.c

```diff
@@ -194,7 +197,13 @@ void xf_SetWindowFullscreen(xfContext* xfc, xfWindow* window, BOOL fullscreen)
 		startY += xfc->context.settings->MonitorLocalShiftY;
 	}
 
-	if (xfc->_NET_WM_FULLSCREEN_MONITORS != None)
+	/*
+	  If window manager has multi-monitor fullscreen support or (even if it doesn't)
+	  a single monitor is being used (either only 1 available or user did not
+	  specified /multimon on command line) it is safe to proceed with simply
+	  toogling _NET_WM_STATE_FULLSCREEN window state.
+	 */
+	if (xfc->_NET_WM_FULLSCREEN_MONITORS != None || settings->MonitorCount == 1)
```

Comment:
This looks like something is wrong. If the WM does provide `_NET_WM_FULLSCREEN_MONITORS` fine, but if it does not and expects it, it is broken. File a bug report with them.